src/main/scala/com/target/data_validator/validator/ColumnBased.scala

```diff
@@ -35,26 +35,40 @@ abstract class ColumnBased(column: String, condTest: Expression) extends CheapCh
   }
 }
 
-case class MinNumRows(minNumRows: Long) extends ColumnBased("", ValidatorBase.L0) {
+case class MinNumRows(minNumRows: Json) extends ColumnBased("", ValidatorBase.L0) {
   override def name: String = "MinNumRows"
 
-  override def substituteVariables(dict: VarSubstitution): ValidatorBase = this
+  override def substituteVariables(dict: VarSubstitution): ValidatorBase = {
+    val ret = MinNumRows(getVarSubJson(minNumRows, "minNumRows", dict))
+    getEvents.foreach(ret.addEvent)
+    ret
+  }
 
   override def configCheck(df: DataFrame): Boolean = {
-    if (minNumRows <= 0) {
-      val msg = s"MinNumRows: $minNumRows <= 0"
+
+    def notNaturalNumber(): Unit = {
+      val msg = "minNumRows must be a natural number"
       logger.error(msg)
       addEvent(ValidatorError(msg))
-      failed = true
-      true
-    } else {
-      false
     }
+
+    minNumRows.asNumber match {
+      case Some(jsonNumber) => jsonNumber.toLong match {
+        case Some(x) if x > 0 =>
```

Comment:
This is probably fine. I thought for a moment that it should be `x >= 0` but I don't think it would make sense for a user to check for a minimum of zero rows.

Reply:
As it is now it is consistent with the condition as it was before.  Failure if 'x <= 0'